File: trunk/+hidi/Sensor.py

from abc import abstractmethod, ABCMeta

import WorldTime
import Trajectory

class Sensor(object):
    __metaclass__ = ABCMeta

    @abstractmethod
    def refresh(self, traj):
        pass

    @abstractmethod
    def hasData(self):
        return False

    @abstractmethod
    def first(self):
        raise ValueError

    @abstractmethod
    def last(self):
        raise ValueError

    @abstractmethod
    def getTime(self, n):
        return WorldTime.WorldTime()
    

File: trunk/+hidi/Sensor.m
classdef Sensor < handle
  methods (Access = protected, Static = true)
    function this = Sensor()
    end
  end
  
  methods (Access = public, Abstract = true)
    refresh(this, varargin);
    flag = hasData(this);
    node = first(this);
    node = last(this);
    time = getTime(this, node);
  end
end
